### PR TITLE
Add a regression test for `learn-ocaml build'

### PR DESCRIPTION
**Pull request by agrn**
Commit 5a661e4 ("repo: fix repo generation when an exercise has multiple subdirectories", 2020-05-26) fixed the issue #305, but it did not add any regression test to the test suite to ensure that the behaviour of `learn-ocaml build' stayed correct.

This series cleans up a bit `runtests.sh`, adds such a test, and makes Travis to check `learn-ocaml build' against learn-ocaml-corpus.

This patch series keeps the general behaviour of `runtests.sh` -- perhaps in the future, it could be changed to have a [TAP](https://testanything.org/)-compliant output, or allow it to run without docker?

### CODE REVIEW

**Comment by agrn** [comment]
Thank you for your feedback @erikmd!

> 
> LGTM, one remark though: there are several occurrences of the pattern
> 
> ```
> for CORPUS in $(find ./corpuses -mindepth 1 -maxdepth 1 -type d) ; do …; done
> ```
> but this won't work if ever there is a folder that contains a space (I know this is a priori not the case, but better safe than sorry)
> 
> so as suggested in https://mywiki.wooledge.org/BashFAQ/024#Workarounds you may want to phrase your loops like this:
> 
> ```
> while IFS= read -r CORPUS; do …; done < <(find ./corpuses -mindepth 1 -maxdepth 1 -type d)
> ```
> 

Eww, as if I needed to be reminded why I hate shell scripts with a
burning passion.  :)

I am fixing this as I am writing these lines.

File: tests/runtests.sh [hunk 1]
```diff
@@ -42,86 +42,112 @@ learn-ocaml --sync-dir=/sync --repo=/repository build serve")
...
-for DIR in `ls -d */`
+while IFS= read -r dir;
 do
     run_server
 
-    pushd $DIR > /dev/null
+    pushd "$dir" > /dev/null
 
-    echo "---> Doing $DIR:"
+    echo "---> Doing $dir:"
```
Comment (yurug) [inline tests/runtests.sh:124]:
```suggestion
    echo "---> Entering $dir:"
```

**Comment by yurug** [comment]
This looks great to me.

Did you try to run [shellcheck](https://github.com/koalaman/shellcheck) on your script?

**Comment by agrn** [comment]
Yes, it reports the usual errors with `pushd`/`popd`, `let`, and backslashes in strings.  Other than that, there is also the only for loop that I did not convert -- for some reason, when converted to a while loop, it will execute only for the first file found.  I am not sure why it behaves like this.

**Comment by yurug** [comment]
Thanks!

**Pull request merged**